client/src/pages/Login.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from "react";
+import { TextInputField } from "./../components/Components";
+import { Container, Grid, Typography, Button } from "@material-ui/core";
+import { IdentityToken, emptyIdentityToken } from "./../interfaces";
+import { extractIdentityToken } from "./../utils";
+import { TITLE_TEXT, microapps, isWeb, isDebug } from "./../constants/index";
+import Logo from "./../img/Logo.png";
+import "./../css/Login.css";
+declare const window: any;
+
+function Login() {
+  const logoSpinnerSpeed = 3;
+
+  // Flag for us to manually run login on mobile
+  const [loginError, setLoginError] = useState(false);
+  // User Identity
+  const [identity, setIdentity] = useState<IdentityToken>(emptyIdentityToken());
+
+  const updateUser = (text: string) => {
+    const newIdentity = emptyIdentityToken();
+    newIdentity.sub = text;
+    setIdentity(newIdentity);
```

Comment:
```suggestion
    setIdentity(Object.assign({},newIdentity,{sub: text}));
```
Reason: preserves immutability. also wondering what does "sub" mean here?

Reply:
Sub is a unique ID associated to each google account which is recommended for applications to use to identify users who logged in via their google account.